src/main/java/seedu/taassist/logic/commands/GradeCommand.java

```diff
@@ -0,0 +1,151 @@
+import seedu.taassist.logic.parser.ParserStudentIndexUtil;
+import seedu.taassist.logic.parser.exceptions.ParseException;
+import seedu.taassist.model.Model;
+import seedu.taassist.model.moduleclass.ModuleClass;
+import seedu.taassist.model.session.Session;
+import seedu.taassist.model.session.SessionData;
+import seedu.taassist.model.student.Student;
+import seedu.taassist.model.student.StudentModuleData;
+import seedu.taassist.model.student.StudentSessionData;
+import seedu.taassist.model.uniquelist.UniqueList;
+
+/**
+ * Gives a grade to student for a session.
+ */
+public class GradeCommand extends Command {
+
+    public static final String COMMAND_WORD = "grade";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Gives a grade to a student for a session. "
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_SESSION + "SESSION (must be a valid session)\n"
+            + PREFIX_GRADE + "GRADE (must be a valid grade) "
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_SESSION + "Tutorial1 "
+            + PREFIX_GRADE + "100";
+
+
+    public static final String MESSAGE_SUCCESS = "Grade given to student: %1$s";
+    public static final String MESSAGE_INVALID_SESSION = "The session provided is invalid: %1$s";
+
+    private final Index index;
+    private final Session session;
+    private final double grade;
+
+    /**
+     * Creates a GradeCommand to give the specified {@code grade} to the student at the specified {@code index}
+     * for the specified {@code session}.
+     */
+    public GradeCommand(Index index, Session session, double grade) {
+        requireAllNonNull(index, session);
+        this.index = index;
+        this.session = session;
+        this.grade = grade;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        if (!model.isInFocusMode()) {
+            throw new CommandException(String.format(MESSAGE_NOT_IN_FOCUS_MODE, COMMAND_WORD));
+        }
+
+        ModuleClass moduleClass = model.getFocusedClass();
+        if (!moduleClass.hasSession(session)) {
+            throw new CommandException(String.format(MESSAGE_INVALID_SESSION, session));
+        }
+
+        List<Student> lastShownList = model.getFilteredStudentList();
+        Student oldStudent;
+        try {
+            oldStudent = ParserStudentIndexUtil.parseStudentFromIndex(index, lastShownList);
+        } catch (ParseException e) {
+            throw new CommandException(String.format(MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX));
+        }
+
+        Student newStudent = updatedStudent(oldStudent, moduleClass, session, grade);
+
+        model.setStudent(oldStudent, newStudent);
+        return new CommandResult(String.format(MESSAGE_SUCCESS, newStudent));
+    }
+
+    private Student updatedStudent(Student oldStudent, ModuleClass moduleClass, Session session, double grade) {
```

Comment:
Should the `update*` methods be moves into the class that's being edited as static methods? It is likely that these are used in multiple places.